### DISCUSSION TITLE
Application recovery and restart APIs

### DISCUSSION THREAD

**Discussion by AlinaSiembor** [microsoft/WindowsAppSDK#3099] [Ideas]
Adding these APIs (https://learn.microsoft.com/en-us/windows/win32/recovery/application-recovery-and-restart-portal) to Windows App SDK would make it much easier to support automatic recovery post device or app reboot. 

Registering for these types of restarts is good practice for any Windows app and current experience through Win32-only APIs requires extra steps in some cases (like C#\native code).

I suggest we bring the following to the App SDK:
      ApplicationRecoveryFinished
      ApplicationRecoveryInProgress
      GetApplicationRecoveryCallback
      GetApplicationRestartSettings
      RegisterApplicationRecoveryCallback
      RegisterApplicationRestart
      UnregisterApplicationRecoveryCallback
      UnregisterApplicationRestart

**Comment by dhoehna** [comment]
> would make it much easier to support automatic recovery post device or app reboot.
I don't see how moving these APIs to Windows App SDK would make ARR easier.  

Maybe we could group related functions together to make ARR easier to use, or Windows App SDK could handle the business rules of ARR.  The remarks for [RegisterApplicationRecoveryCallback](https://learn.microsoft.com/en-us/windows/win32/api/winbase/nf-winbase-registerapplicationrecoverycallback) say `You must call the [ApplicationRecoveryInProgress](https://learn.microsoft.com/en-us/windows/desktop/api/winbase/nf-winbase-applicationrecoveryinprogress) function within the specified interval to indicate to ARR that you are still actively recovering;` simple thing to forget.  Windows App SDK could handle this.
